src/clj/puppetlabs/services/master/master_service.clj

```diff
@@ -79,6 +79,25 @@
    default-jvm-metrics-allowed
    http-client-metrics-allowed-hists))
 
+(defn jdk-support-status
+  "Returns :official, :deprecated, :unknown, or :unsupported."
+  [version]
+  (cond
+    (re-matches #"1\.[1234567]($|(\..*))" version) :unsupported
+    (re-matches #"1\.[89]($|(\..*))" version) :deprecated
+    (re-matches #"10($|(\..*))" version) :deprecated
+    (re-matches (re-pattern (str 11 "($|(\\..*))")) version) :official
+    (re-matches #"17($|(\..*))" version) :official
+    :else :unknown))
+
+(defn log-java-deprecation-message [version]
+  (let [status (jdk-support-status version)]
+    (case status
+      (:unknown) (log/warn (i18n/trs "JDK {0} is neither tested nor supported. Please use JDK 11 or 17" version))
+      (:deprecated) (log/warn (i18n/trs "JDK {0} is deprecated, please upgrade to JDK 11 or 17" version))
+      (:official) nil
+      (log/warn (i18n/trs "Puppetserver doesn't support JDK {0}" version)))))
```

Comment:
I think this should be `:unsupported` since we always return one of four statuses.